lib/loadGltfUris.js

```diff
@@ -94,7 +98,12 @@ function loadURI(gltf, basePath, name, callback) {
     }
 }
 
+//Return the extension of the data uri
  function getImageDataUriFormat(uri) {
-    //Return the extension of the data uri
-    return '.' + uri.match('\/([^)]+)\;')[1];
+    var extension = uri.match('\/([^)]+)\;');
+    if (extension === null) {
```

Comment:
Ah, I see.

So we're assuming .png for binary glTF...for example if someone used gltf-pipeline to convert from binary glTF to glTF with separate resources, this would fail if the images were actually jpg?

I _think_ we're going to have to inspect the header of the binary blob to determine the image format (it can only be `jpg`, `png`, `bmp`, or `gif`).

Go for it now if this interests you; otherwise, submit an issue with a link to this discussion and we can consider it in a bit.